presto-main/src/test/java/com/facebook/presto/operator/TestHashJoinOperator.java

```diff
@@ -677,7 +680,19 @@ private TaskContext createTaskContext()
     private static LookupSourceFactory buildHash(boolean parallelBuild, TaskContext taskContext, List<Integer> hashChannels, RowPagesBuilder buildPages, Optional<InternalJoinFilterFunction> filterFunction)
     {
         Optional<JoinFilterFunctionFactory> filterFunctionFactory = filterFunction
-                .map(function -> ((session, addresses, channels) -> new StandardJoinFilterFunction(function, addresses, channels)));
+                .map(function -> new JoinFilterFunctionFactory() {
+                    @Override
+                    public JoinFilterFunction create(ConnectorSession session, LongArrayList addresses, List<List<Block>> channels)
+                    {
+                        return new StandardJoinFilterFunction(function, addresses, channels, Optional.empty());
+                    }
+
+                    @Override
+                    public Optional<SortChannelExtractor.SortChannel> getSortChannel()
+                    {
+                        return Optional.empty();
```

Comment:
I didn't want to spend a lot of time adding tests without initial approval that this thing will go into code base.